metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart

```diff
@@ -141,6 +141,28 @@ void main() {
       },
     );
 
+    test(
+      ".headers contains the given additional headers",
+      () {
+        final additionalHeaders = {
+          HttpHeaders.userAgentHeader: null,
+        };
+
+        final client = BuildkiteClient(
+          buildkiteApiUrl: buildkiteMockServer.url,
+          organizationSlug: organizationSlug,
+          authorization: authorization,
+          headers: additionalHeaders,
+        );
+
+        final headers = client.headers;
+
+        final expectedHeader = additionalHeaders.entries.first;
+
+        expect(headers, containsPair(expectedHeader.key, expectedHeader.value));
```

Comment:
```suggestion
        const expectedKey = 'test-header';
        const expectedValue = 'test-value';
        const additionalHeaders = {
          expectedKey: expectedValue,
        };

        final client = BuildkiteClient(
          buildkiteApiUrl: buildkiteMockServer.url,
          organizationSlug: organizationSlug,
          authorization: authorization,
          headers: additionalHeaders,
        );
        final headers = client.headers;
        
        expect(headers, containsPair(expectedKey, expectedValue));
```
Please apply to similar test cases.